packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -329,8 +372,27 @@ export class EngineReportingAgent<TContext = any> {
     handleLegacyOptions(this.options);
   }
 
+  public executableSchemaIdGenerator(schema: string | GraphQLSchema) {
+    if (this.lastSeenExecutableSchemaToId?.executabeSchema === schema) {
+      return this.lastSeenExecutableSchemaToId.executableSchemaId
+    }
+    const id = computeExecutableSchemaId(schema);
+
+    this.lastSeenExecutableSchemaToId = {
+      executabeSchema: schema,
+      executableSchemaId: id,
+    }
+
+    return id;
+  }
+
   public newPlugin(): ApolloServerPlugin<TContext> {
-    return plugin(this.options, this.addTrace.bind(this));
+    return plugin(
+      this.options,
+      this.addTrace.bind(this),
+      this.startSchemaReporting.bind(this),
+      this.executableSchemaIdGenerator.bind(this),
```

Comment:
Seems reasonable to pass a second object with these grouped together unless the positional parameters seem ideal.  In terms of the `apollo-engine-reporting` API, it's already getting a "major" version update, so the API is free for the changing right now (previously we passed in `schemaHash` in this second position).

Anyhow, not a hard ask.  (Also see a somewhat-related future-thinking comment about rolling these two things together in https://github.com/apollographql/apollo-server/pull/3998/commits/de7ba72b50ff539fc33b59cf88553527b6e99a70#r414904554.  Again, no reason to do that today.)